IHub-224047 Add ability to upload files using bytes instead of file path

ov-automation opened the pull request:
[IHub-224047]( https://trackor.onevizion.com/form/ConfigAppForm.do?id=100110413245&ttid=10009161) Add ability to upload files using bytes instead of file path<br><br>[IHub-224047-129754](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100110691108&ttid=100091445) vadim.glebov (Vadim Glebov)<br>[IHub-224047-129756](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100110691109&ttid=100091445) amoiseenko (Andrey Moiseenko)

Anastasiyaaa committed:
IHub-224047 Add ability to upload files using bytes instead of file path
IHub-224047-127863 [DEV] Add ability to upload files using bytes instead of file path

glebowadim commented on onevizion/trackor.py:
```diff
@@ -484,26 +484,28 @@ def get_filename_from_cd(cd):
 
 
-	def UploadFile(self, trackorId, fieldName, fileName, newFileName=None):
+	def UploadFile(self, trackorId, fieldName, fileName, newFileName=None, byteString=None):
```
Propose to make ```fileName``` optional, and use only ```newFileName``` as filename if ```byteString``` is present

Anastasiyaaa replied:
Done

Anastasiyaaa committed:
IHub-224047 Add ability to upload files using bytes instead of file path
IHub-224047-129754 [CR] Add ability to upload files using bytes instead of file path

glebowadim commented on onevizion/trackor.py:
```diff
@@ -512,7 +527,7 @@ def UploadFile(self, trackorId, fieldName, fileName, newFileName=None):
 
 		Message(URL,2)
-		Message("FileName: {FileName}".format(FileName=fileName),2)
+		Message("FileName: {FileName}".format(FileName=newFileName if byteString else fileName),2)
```
I think we need to log both the filename and the file path.

In this implementation, either the name or the path is logged. This is weird

Anastasiyaaa replied:
Done

glebowadim commented on onevizion/trackor.py:
```diff
@@ -484,26 +484,41 @@ def get_filename_from_cd(cd):
 		"""
 
+		if not fileName and not byteString:
+			self.errors.append("File path (fieldName) or byteString must be passed to the method")
```
```Either "fileName" or "newFileName" and "byteString" are required```

Anastasiyaaa replied:
Fixed

Anastasiyaaa committed:
IHub-224047 Add ability to upload files using bytes instead of file path
IHub-224047-129754 [CR] Add ability to upload files using bytes instead of file path

asmoiseenko commented on onevizion/trackor.py:
```diff
@@ -484,26 +484,38 @@ def get_filename_from_cd(cd):
 
 
-	def UploadFile(self, trackorId, fieldName, fileName, newFileName=None):
+	def UploadFile(self, trackorId, fieldName, fileName=None, newFileName=None, byteString=None):
```
I propose a separate oveloaded procedure

Anastasiyaaa replied:
Done without overload

Anastasiyaaa committed:
IHub-224047 Add ability to upload files using bytes instead of file path
IHub-224047-129756 [CR] Add ability to upload files using bytes instead of file path

asmoiseenko commented on onevizion/trackor.py:
```diff
@@ -487,23 +487,38 @@ def get_filename_from_cd(cd):
+			trackorID: the system ID for the particular Trackor record that this is being assigned to.
+			fieldName: should be the Configured Field Name, not the Label.
+			fileName: name of the file you want to upload.
+			fileContents: byte string or BufferdReadeer of the file you want to upload.
```
fix typo in the BufferdRead**ee**r

Anastasiyaaa replied:
Fixed

asmoiseenko commented on onevizion/trackor.py:
```diff
@@ -487,23 +487,38 @@ def get_filename_from_cd(cd):
+
+
+	def UploadFileByFileContents(self, trackorId, fieldName, fileName, fileContents):
+		""" Get a File from a particular Trackor record's particular Configured field
```
Why the comment says `Get a File`? Shouldn't is say `Upload a file to a ...`

Anastasiyaaa replied:
Fixed

Anastasiyaaa committed:
IHub-224047 Add ability to upload files using bytes instead of file path
IHub-224047-129756 [CR] Add ability to upload files using bytes instead of file path
